capture.cpp

```diff
@@ -1506,15 +1505,43 @@ const char *locale = DEFAULT_LOCALE;
         printf("  - Camera with cooling capabilities\n");
     }
 
-    const char *ver = ASIGetSDKVersion();
-    printf("  - SDK version %s\n", ver);
-
-    asiRetCode = ASIInitCamera(CamNum);
-    if (asiRetCode == ASI_SUCCESS)
+    printf("\n");
+    ASI_ID cameraID;	// USB 3 cameras only
+    if (ASICameraInfo.IsUSB3Camera == ASI_TRUE && ASIGetID(CamNum, &cameraID) == ASI_SUCCESS)
+    {
+        printf("  - Camera ID: ");
+        if (cameraID.id[0] == '\0')
+        {
+            printf("[none]");
+        } else {
+            for (unsigned int i=0; i<sizeof(cameraID.id); i++) printf("%c", cameraID.id[i]);
+        }
+        printf("\n");
+    }
+    // To clear the camera ID:
+        // cameraID.id[0] = '\0';
+        // ASISetID(CamNum, cameraID);
+    ASI_SN serialNumber;
+    asiRetCode = ASIGetSerialNumber(CamNum, &serialNumber);
+    if (asiRetCode != ASI_SUCCESS)
     {
-        printf("  - Initialise Camera OK\n");
+        printf("*** WARNING: unable to get serialNumber (%s)\n", getRetCode(asiRetCode));
     }
     else
+    {
+        printf("  - Camera Serial Number: ");
+        if (serialNumber.id[0] == '\0')
+        {
+            printf("[none]");
+        } else {
+            printf("0x");

```

Comment:
serial number is 8 bytes, which just happens to be the size of a `long long` or `uint64_t`.  You could do something like this.

```
} else {
      uint64_t sn;
      memcpy(&sn, &serialNumber, sizeof(sn));
      printf("0x%016lx", sn);
}
```